.circleci/config.yml

```diff
@@ -1,53 +1,54 @@
-# Python CircleCI 2.0 configuration file
-#
-# Check https://circleci.com/docs/2.0/language-python/ for more details
-#
 version: 2
-machine:
-  python:
-    version: pypy-2.7.1
 jobs:
   build:
-    docker:
-      # specify the version you desire here
-      # use `-browsers` prefix for selenium tests, e.g. `3.6.1-browsers`
-      - image: circleci/python:2.7
-      
-      # Specify service dependencies here if necessary
-      # CircleCI maintains a library of pre-built images
-      # documented at https://circleci.com/docs/2.0/circleci-images/
-      # - image: circleci/postgres:9.4
 
     working_directory: ~/repo
 
+    machine:
+      python:
+        version: pyenv 2.7.12
+      enabled: true
+
     steps:
-      - checkout
-
-      # Download and cache dependencies
-      - restore_cache:
-          keys:
-          - v1-dependencies-{{ checksum "requirements.txt" }}
-          # fallback to using the latest cache if no exact match is found
-          - v1-dependencies-
-
-      - run:
-          name: install dependencies
-          command: |
-            virtualenv .venv/bin/activate
-            sudo pip install -r requirements.txt
-
-      - save_cache:
-          paths:
-            - ./venv
-          key: v1-dependencies-{{ checksum "requirements.txt" }}
-        
-      # run tests!
-      - run:
-          name: run tests
-          command: |
-            sudo -H python test_main.py
-
-      - store_artifacts:
-          path: test-reports
-          destination: test-reports
-          
+        - checkout
+
+        # Restore venv dependencies
+        - restore_cache:
+            keys:
+            - v1-dependencies-{{ checksum "requirements.txt" }}
+            # fallback to using the latest cache if no exact match is found
+            - v1-dependencies-
+
+        # Restore non-python dependencies
+        - restore_cache:
+            keys:
+            - v1-dependencies-{{ checksum "base_installation.sh" }}
+            - v1-dependencies-
+
+        - run:
+            name: Install dependencies
+            command: |
+              virtualenv .venv/bin/activate
+              sudo apt-get install build-essential python-dev curl file git python-setuptools ruby
+        - run:
+            name: run tests
+            command: |
+              mkdir -p ~/.local/bin/
+              export TERM=xterm
+              ./base_installation.sh
```

Comment:
Move the ```base_install.sh``` call into the install dependancies run command. Below the sudo apt-get install build-essential python-dev curl file git python-setuptools ruby